src/circuit/circuit_frame.rs

```diff
@@ -1216,12 +1233,23 @@ fn reduce_cons<F: PrimeField, CS: ConstraintSystem<F>>(
 
         let val = AllocatedPtr::pick(
             &mut cs_letrec.namespace(|| "pick val"),
-            &bindings_is_nil,
+            &cond_body1,
```

Comment:
I don't follow this at all. I *think* that if `rest_body` is not nil, that should be an error. But this seems to be saying in that case `val` will be `body1`. Why? Won't this be irrelevant since there is an error? What purpose does `cond_body1` serve?

Reply:
This is the condition for returning body as output expression. This is the behavior in eval.rs. I need to improve the name indeed.